### ISSUE TITLE
Allow configurable tasks, and allow you to specify them on the command line

### ISSUE THREAD

**Issue by jpnelson** [treshugart/gulp-auto-task#2]
API would be: you can export a function that takes a config as one of its arguments, and `gulp-auto-task` will let you specify these options on the command line.

**Comment by treshugart** [comment]
I agree that this would be awesome. How do you envision passing the options from the command line? I was thinking that it might be good as a configuration option, to have a function that would return the options and it would default to using [minimist](https://github.com/substack/minimist) to retrieve them. That would allow the user to configure them.

The only caveat I see to that is that it will be passing in the same options to each factory that will return the Gulp tasks. Maybe this isn't so bad. It removes coupling from having a shared arguments module (i.e. having a shared Commander definition that we get args from). But if you need to configure a gulp task with specific arguments, then this wouldn't solve that. Maybe if you're doing that, then you just call `gulp.task()` manually. This would probably be an edge case since most tasks probably won't use the arguments passed in anyways.
